src/RolloutScorer/RolloutScorer.Tests/RolloutScorerTests.cs

```diff
@@ -0,0 +1,96 @@
+        private const string URI_DOMAIN = "azure.com";
+
+        private const string EXPECTED_GOOD_REDIRECT_MESSAGE = "Response status code does not indicate success: 401";
+        private const string EXPECTED_BAD_HOST_REDIRECT_MESSAGE = "Bad redirect host";
+        private const string EXPECTED_BAD_SCHEME_REDIRECT_MESSAGE = "Bad redirect scheme";
+
+        private RolloutScorer _rolloutScorer;
+
+        public static IEnumerable<object[]> GetUriTestCases()
+        {
+            return new List<object[]>
+            {
+                new object[] { "same-host-redirect", $"https://{URI_SUBDOMAIN}{URI_DOMAIN}/dnceng/redirected", EXPECTED_GOOD_REDIRECT_MESSAGE },
+                new object[] { "different-subdomain-redirect", $"https://different.{URI_DOMAIN}/redirect", EXPECTED_BAD_HOST_REDIRECT_MESSAGE },
+                new object[] { "no-subdomain-redirect", $"https://{URI_DOMAIN}/redirect", EXPECTED_BAD_HOST_REDIRECT_MESSAGE },
+                new object[] { "different-host-redirect", $"https://evil.com/redirect", EXPECTED_BAD_HOST_REDIRECT_MESSAGE },
+                new object[] { "no-https-redirect", $"http://{URI_SUBDOMAIN}{URI_DOMAIN}/dnceng/redirected", EXPECTED_BAD_SCHEME_REDIRECT_MESSAGE },
+            };
+        }
+
+        public RolloutScorerTests()
+        {
+            _rolloutScorer = new RolloutScorer();
+            Config config = Utilities.ParseConfig();
+            _rolloutScorer.RolloutWeightConfig = config.RolloutWeightConfig;
+            _rolloutScorer.RepoConfig = config.RepoConfigs.First();
+            _rolloutScorer.AzdoConfig = config.AzdoInstanceConfigs.Find(a => a.Name == _rolloutScorer.RepoConfig.AzdoInstance);
+            _rolloutScorer.Repo = _rolloutScorer.RepoConfig.Repo;
+            _rolloutScorer.RolloutStartDate = DateTimeOffset.Now.AddDays(-1);
+
+            _rolloutScorer.SetupHttpClient(new Microsoft.Azure.KeyVault.Models.SecretBundle(value: "fakePat"));
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUriTestCases))]
+        public async Task RedirectApiTest(string requestPath, string responseUri, string expectedMessage)
+        {
+            Uri sameHostRedirectUri = new Uri($"https://{URI_SUBDOMAIN}{URI_DOMAIN}/dnceng/{requestPath}");
+            HttpResponseMessage sameHostRedirectResponse = new HttpResponseMessage();
+            sameHostRedirectResponse.Headers.Location = new Uri(responseUri);
+
+            try
+            {
+                await _rolloutScorer.HandleApiRedirect(sameHostRedirectResponse, sameHostRedirectUri);
+            }
+            catch (HttpRequestException e)
+            {
+                Assert.Contains(expectedMessage, e.Message);
+            }
+        }
+
+        // We're only testing Octokit because if this behavior changes, we won't know in our code and our scorecard results will be silently incorrect
+        [Fact]
+        public async Task OctokitReturnsOpenAndClosedIssuesTest()
```

Comment:
I REALLY don't like testing every random third party library we depend on.  That's a huge testing burden... we don't duplicate all of the tests for System.String, and we'd be just as broken if that happened.

We are going to need a super solid reason this test exists other than "we depend on this behavior", otherwise we should delete this test. And it definitely shouldn't be in a class called RolloutScorerTests, since it doesn't even create one.

Reply:
It also depends on external resources, which ALSO means it's flaky, two reasons not to check it in, and at the very least it needs to somehow be "OuterLoop" if we are going to be actually contacting github in the test (though, again, I think it shouldn't exist at all, especially with no mocking of any kind)

Reply:
I'll remove it; I'm confident we'll notice failures if we're paying attention to the scorecards now.